File: Java/Non-Constructable_Change.java
package Java;

import java.util.*;

// Time: O(n log(n))
// Space: O(1)
class Program {

    public int nonConstructibleChange(int[] coins) {
        Arrays.sort(coins);
        int totalCoins = 0;

        for (int i : coins) {
            if (i > totalCoins + 1) {
                return totalCoins + 1;
            }
            totalCoins += i;
        }
        return totalCoins + 1;
    }
}

File: Python/Non-Constructable_Change.py
# Time: O(n log(n))
# Space: O(1)
def nonConstructibleChange(coins):
    coins.sort()
    totalCoins = 0

    for i in coins:
        if i > totalCoins + 1:
            return totalCoins + 1

        totalCoins += i

    return totalCoins + 1
